internal/controller/management_controller.go

```diff
@@ -306,9 +306,17 @@ func (r *ManagementReconciler) cleanupRemovedComponents(ctx context.Context, man
 
 		componentName := hr.Name // providers(components) names map 1-1 to the helmreleases names
 
+		releasesList := &metav1.PartialObjectMetadataList{}
+		releasesList.SetGroupVersionKind(kcm.GroupVersion.WithKind(kcm.ReleaseKind))
+		if err := r.Client.List(ctx, releasesList); err != nil {
+			return fmt.Errorf("failed to list releases: %w", err)
+		}
```

Comment:
i believe we could call the `list` just once before the loop

Reply:
Oh, right. Thanks.

Reply:
and just in case, i'd call the `list` only if `len(managedHelmReleases.Items) > 0`